tutorials/ExploringDeutschJozsaAlgorithm/DeutschJozsaAlgorithmTutorial.ipynb

```diff
@@ -620,7 +620,83 @@
    "cell_type": "markdown",
    "metadata": {},
    "source": [
-    "# Part V. What's Next?\n",
+    "# Part V.  Different function types - what if our function is neither balanced nor constant?\n",
+    "\n",
+    "The setup for the Deutsch-Josza algorithm presupposes that the function we're trying to identify is perfectly constant or perfectly balanced in its output (indeed, we haven't even considered the possibility that an input outside of this space could be used). Although the algorithm will not converge to a solution in the same manner that we've seen so far when selecting another type of function as input, we can still derive some intuition for how quantum states behave probabilistically by considering some outside-the-box cases.\n",
+    "\n",
+    "Before thinking about particular cases, it's worthwhile to consider how to classify (in a purely qualitative fashion) our function $f(x)$ with respect to its outputs being balanced or constant. One might consider $f(x)$ to be a superposition of balanced and constant functions, such that it conveys incomplete properties of both types. $f(x)$ could then be said to be balanced for some subset of its inputs and constant for the remainder. We could ask ourselves, \"How balanced is $f(x)$?\" in the same way that we might ask, for a qubit, \"How 'close' is the qubit to a $|0\\rangle$ or $|1\\rangle$ state?\"\n",
+    "\n",
+    "For example, consider the following oracle that introduces an unbalanced (read: 'constant') aspect into what you might recognize as the operation for `PhaseOracle_Xmod2`:"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 1,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "operation PhaseOracle_AlmostBalanced (x : Qubit[]) : Unit {\n",
+    "     // Supply a -1 phase to the last qubit's |1> basis state\n",
+    "     Z(x[Length(x) - 1]);\n",
+    "     \n",
+    "     // Subject to a tight set of controls, supply an additional -1 phase to a limited set of states as follows:\n",
+    "     // -Define the controls for a Controlled Z gate as all but the top two qubits\n",
+    "     // -Apply the phase to the qubit immediately below the MSB\n",
+    "     // -The net result, using an example of 8 qubits, will affect two states:\n",
+    "     //    |01111111>\n",
```

Comment:
I would suggest using a more convenient big-endian notation here; earlier in the tutorial (in the oracles section) we use it to say that least significant bit corresponds to the last element of the array, so it makes sense to stick to the same convention here.

Reply:
This should be taken care of by the change in the function - we're only left with the state |11111111> now.